### PR TITLE
Add high level metrics explanation docs page

### PR DESCRIPTION
**Pull request by david-martin**
Closes https://github.com/Kuadrant/docs.kuadrant.io/issues/74

### CODE REVIEW

**Comment by david-martin** [comment]
Updated to include metrics and telemetry config for latest dashboards.
Moved from draft.

**Comment by jasonmadigan** [comment]
👀

File: doc/observability/metrics.md [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+        metric: REQUEST_DURATION
+      tagOverrides:
+        request_url_path:
+          value: "request.url_path"
+```
+
+## State metrics
+
+The [kube-state-metrics](https://github.com/kubernetes/kube-state-metrics/tree/main/docs#default-resources)
+project exposes the state of various kuberenetes resources
```
Comment (jasonmadigan) [inline doc/observability/metrics.md:71]:
```suggestion
project exposes the state of various Kubernetes resources
```

File: doc/observability/metrics.md [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+as metrics and labels. For example, the ready `status` of a `Pod` is available as
+`kube_pod_status_ready`, with labels for the pod `name` and `namespace`. This can
+be useful for linking lower level container metrics back to a meaningful resource
+in the kubernetes world.
+
+## Joining metrics
+
+Metric queries can be as simple as just the name of the metric, or can be complex
+with joining & grouping. A lot of the time it can be useful to tie back low level
+metrics to more meaningful kubernetes resources. For example, if the memory usage
```
Comment (jasonmadigan) [inline doc/observability/metrics.md:81]:
```suggestion
metrics to more meaningful Kubernetes resources. For example, if the memory usage
```

File: doc/observability/metrics.md [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+is maxed out on a container and that container is constantly being OOMKilled, it
+can be useful to get the Deployment and Namespace of that container for debugging.
+Prometheus query language (or promql) allows [vector matching](https://prometheus.io/docs/prometheus/latest/querying/operators/#vector-matching)
+or results (sometimes called joining).
+
+When using Gateway API and Kuadrant resources like HTTPRoute and RateLimitPolicy,
+the state metrics can be joined to istio metrics to give a meaningful result set.
```
Comment (jasonmadigan) [inline doc/observability/metrics.md:88]:
```suggestion
the state metrics can be joined to Istio metrics to give a meaningful result set.
```

File: doc/observability/metrics.md [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+        istio_requests_total{}[5m]
+    )
+) by (destination_service_name)
+
+* on(destination_service_name) group_right 
+    label_replace(gatewayapi_httproute_labels{}, \"destination_service_name\", \"$1\",\"service\", \"(.+)\")
+```
+
+Breaking this query down, there are 2 parts.
+The first part is getting the rate of requests hitting the istio gateway, aggregated
```
Comment (jasonmadigan) [inline doc/observability/metrics.md:104]:
```suggestion
The first part is getting the rate of requests hitting the Istio gateway, aggregated
```

File: doc/observability/metrics.md [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+```
+
+The result set here will include a label for the destination service name (i.e.
+the Service in Kubernetes). This label is key to looking up the HTTPRoute this
+traffic belongs to.
+
+The 2nd part of the query uses the `gatewayapi_httproute_labels` metric and the
+`label_replace` function. The `gatewayapi_httproute_labels` metric gives a list
+of all httproutes, including any labels on them. The HTTPRoute in this example
+has a label called 'service', set to be the same as the istio service name.
```
Comment (jasonmadigan) [inline doc/observability/metrics.md:122]:
```suggestion
has a label called 'service', set to be the same as the Istio service name.
```

File: doc/observability/metrics.md [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+
+```promql
+* on(destination_service_name) group_right 
+```
+
+* `*` is the binary operator i.e. multiplication (gives join like behaviour)
+* `on()` specifies which labels to "join" the 2 results with
+* `group_right` enables a one to many matching.
+
+See the [prometheus documentation](https://prometheus.io/docs/prometheus/latest/querying/operators/#vector-matching) for further details on matching.
```
Comment (jasonmadigan) [inline doc/observability/metrics.md:142]:
```suggestion
See the [Prometheus documentation](https://prometheus.io/docs/prometheus/latest/querying/operators/#vector-matching) for further details on matching.
```

**Review by jasonmadigan — APPROVED** [review]
Couple of small tweaks as suggestions, but otherwise looks good

**Review by jasonmadigan — APPROVED** [review]
Couple of small tweaks as suggestions, but otherwise looks good

**Comment by david-martin** [comment]
Thanks @jasonmadigan 
Suggested accepted & squashed.

**Comment by jasonmadigan** [comment]
Looks good, merge when built

**Pull request merged**